datahub-frontend/app/controllers/api/v2/Dataset.java

```diff
@@ -264,4 +267,15 @@ public Result getDatasetDownstreams(@Nonnull String datasetUrn) {
     }
     return ok(Json.toJson(downstreams));
   }
+
+  @Security.Authenticated(Secured.class)
+  @Nonnull
+  public Result getDataPlatforms() {
+    try {
+      return ok(ControllerUtil.jsonNode("platforms", _dataPlatformsDao.getAllPlatforms()));
+    } catch (final Exception e) {
+      Logger.error("Fail to get data platforms", e);
```

Comment:
Why catch and return 404? Why not just leave exceptions so presumably this returns 500?

Reply:
As discussed offline, I will not address this because this is coming from internal codebase as is. We should do this refactoring in the internal codebase first. Just to unblock things in favor of pushing `datahub-web` code, I will accelerate merging this PR.

Reply:
Also, the UI's contract with the midtier is to treat 404's gracefully with an FE provided default value, whereas 500 is a "PANIC EVERYTHING'S GONE WRONG" message

Reply:
404 is not found. 500 is indeed something went wrong. And exception is something went wrong :)